app/models/Search.java

```diff
@@ -487,4 +495,15 @@ private SearchResponse search(final QueryBuilder queryBuilder,
 	private void setMessageOut(final Chunks.Out<String> messageOut) {
 		this.messageOut = messageOut;
 	}
+
+	/**
+	 * As there is a memory issue within Promise we musn't allow data > RAM. As
+	 * experience shows, this is around 4 M docs (dependig on the serialisation).
+	 * 
+	 * @return maximum of allowed hits for a scroll search
+	 */
+
+	public static long getAllowedMaxScrollHits() {
+		return maxScrollHits;
```

Comment:
Instead of a private field and a public static getter, shouldn't this be a normal constant (also `int` is enough here), i.e.
`public static final int MAX_SCROLL_HITS = 3000000`?